packages/graph/hash_graph/lib/graph/src/api/rest/link_type.rs

```diff
@@ -128,23 +132,43 @@ async fn create_link_type<P: StorePool + Send>(
         .map(Json)
 }
 
+#[derive(Deserialize, Component)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+struct LinkTypeQuery {
+    query: Expression,
+}
+
 #[utoipa::path(
     post,
     path = "/link-types/query",
-    request_body = Expression,
+    request_body = LinkTypeQuery,
     tag = "LinkType",
     responses(
-        (status = 200, content_type = "application/json", description = "List of all link types matching the provided query", body = [PersistedLinkType]),
+        (status = 200, content_type = "application/json", description = "List of all link types matching the provided query", body = [LinkTypeTree]),
 
         (status = 422, content_type = "text/plain", description = "Provided query is invalid"),
         (status = 500, description = "Store error occurred"),
     )
 )]
 async fn get_link_types_by_query<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
-    Json(expression): Json<Expression>,
-) -> Result<Json<Vec<PersistedLinkType>>, StatusCode> {
-    read_from_store(pool.as_ref(), &expression).await.map(Json)
+    query: Json<LinkTypeQuery>,
+) -> Result<Json<Vec<LinkTypeTree>>, StatusCode> {
+    let LinkTypeQuery { query } = query.0;
+
+    pool.acquire()
+        .map_err(|error| {
+            tracing::error!(?error, "Could not acquire access to the store");
+            StatusCode::INTERNAL_SERVER_ERROR
+        })
+        .and_then(|store| async move {
+            store.get_link_type(&query).await.map_err(|report| {
+                tracing::error!(error=?report, ?query, "Could not read link type from the store");
```

Comment:
Same comment about multiple types (can you apply to other files, not including a suggestion in this comment for that reason)